assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -1560,6 +1598,23 @@ protected String build_student_view_submission_context(VelocityPortlet portlet,
 
 			// can the student view model answer or not
 			canViewAssignmentIntoContext(context, assignment, s);
+			
+			// add TII info if needed
+			if (allowReviewService && assignment.getContent().getAllowReviewService() && allowLTIReviewService){
+				//put the LTI assignment link in context
+				String ltiLink = contentReviewService.getLTIAccess(currentAssignmentReference, contextString);
+				M_log.debug("ltiLink " + ltiLink);
+				context.put("ltiLink", ltiLink);
+				int factor = AssignmentService.getScaleFactor();
+				int dec = (int)Math.log10(factor);
+				int maxPointsInt = assignment.getContent().getMaxGradePoint() / dec;
```

Comment:
Previous 3 lines are repeated a few times throughout the patch, should be refactored into a method